### ISSUE TITLE
Support for extensions

### ISSUE THREAD

**Issue by davidvc** [civisanalytics/swagger-diff#20]
A colleague added an extension to our Swagger (using the @Extension annotation for Jersey) that looks like this:

    extensions = { @Extension(properties = { @ExtensionProperty(name = "x-cached", value = "true") }) })

This causes swagger-diff to fail:
```
   Hashie::CoercionError: Cannot coerce property "get" from Hash to Swagger::V2::Operation: The property 'x-cached' is not defined for Swagger::V2::Operation.
    /home/jenkins/.rvm/gems/ruby-2.1.2/gems/hashie-3.3.2/lib/hashie/extensions/coercion.rb:46:in `rescue in set_value_with_coercion'
    /home/jenkins/.rvm/gems/ruby-2.1.2/gems/hashie-3.3.2/lib/hashie/extensions/coercion.rb:34:in `set_value_with_coercion'
    /home/jenkins/.rvm/gems/ruby-2.1.2/gems/hashie-3.3.2/lib/hashie/extensions/indifferent_access.rb:107:in `indifferent_writer'
    /home/jenkins/.rvm/gems/ruby-2.1.2/gems/hashie-3.3.2/lib/hashie/dash.rb:168:in `block in initialize_attributes'
    /home/jenkins/.rvm/gems/ruby-2.1.2/gems/hashie-3.3.2/lib/hashie/dash.rb:167:in `each_pair'
    /home/jenkins/.rvm/gems/ruby-2.1.2/gems/hashie-3.3.2/lib/hashie/dash.rb:167:in `initialize_attributes'
    /home/jenkins/.rvm/gems/ruby-2.1.2/gems/hashie-3.3.2/lib/hashie/dash.rb:99:in `initialize'
    /home/jenkins/.rvm/gems/ruby-2.1.2/gems/swagger-core-0.2.3/lib/swagger/swagger_object.rb:16:in `initialize'
```

Is there any chance we could fix this?

Thanks,

David

**Comment by jeffreyc** [comment]
Thank you for your feedback! It looks like this is an issue with [`swagger-rb`](https://github.com/swagger-rb/swagger-rb), the underlying library `swagger-diff` uses to parse Swagger specifications. Unfortunately, we’re juggling a few different priorities at this time, and probably won’t get to this quickly. If you’re interested, we’d welcome a PR addressing this issue. It looks like there are a few possible ways to approach this:

1. File an [issue](https://github.com/swagger-rb/swagger-rb/issues) with `swagger-rb` and wait for them to fix it. However, `swagger-rb` appears to be inactive, or possibly abandoned, so I don’t anticipate any issue being addressed quickly.

2. Add another monkey patch to [patch.rb](https://github.com/civisanalytics/swagger-diff/blob/master/lib/swagger/diff/patch.rb) and a new spec to [specification_spec.rb](https://github.com/civisanalytics/swagger-diff/blob/master/spec/swagger/diff/specification_spec.rb#L181). This is a stopgap, but should be relatively easy.

3. Replace the Swagger parser. [`ruby_swagger`](https://rubygems.org/gems/ruby-swagger) looks to be a viable candidate that's under active development, though it's also pretty young, so may not be feature-complete yet. [`swagger_parser`](https://rubygems.org/gems/swagger_parser) is another candidate. It's less active, though also has no outstanding issues/PRs, so may just be stable. This is better than a monkey patch, but would be a *lot* of work.

If you opt for 2 or 3, you may also want to file an issue with `swagger-rb` so they’re aware and can address the underlying problem if and when they have time.

Either way, we’d be happy to review a PR, and appreciate the contribution!

**Comment by davidvc** [comment]
Thanks!  I'll consider these possibilities.  We, like you, are juggling
priorities :)

**Comment by jeffreyc** [comment]
I just released Swagger::Diff 1.1.0. Among other improvements, it replaces the underlying parser, and should now support your Swagger specification. If you encounter any other problems, please let us know.

**Comment by davidvc** [comment]
That's great, thanks!  I'll give it a shot as soon as I can.
